src/resolve/resolved-dns-server.c

```diff
@@ -94,11 +93,10 @@ int dns_server_new(
                 manager_set_dns_server(m, NULL);
 
         if (server_name) {
-                name = strdup(server_name);
-                if (!name)
+                s->server_name = strdup(server_name);
+                if (!s->server_name)
```

Comment:
Ouch! I notice that this can cause memleak. Sorry. Could you update this like the following?
```
_cleanup_free_ char *name = NULL;

if (server_name)
    name = strdup(server_name);
    if (!name)
        return -ENOMEM;
}

s = new(DnsServer, 1);
if (!s)
    return -ENOMEM;

*s = (DnsServer) {
    ...,
    .server_name = TAKE_PTR(name),
}
```
Otherwise, if `strdup()` fails, then `s` is leaked.